doc/srv6/srv6_policy.md

```diff
@@ -0,0 +1,511 @@
+        },
+        "slist_c": {
+            "10": "fd00:205:205:fff5:55::",
+            "20": "fd00:207:207:fff7:77::"
+        },
+        "slist_d": {
+            "10": "fd00:206:206:fff6:66::",
+            "20": "fd00:204:204:fff4:44::"
+        }
+    },
+```
+
+# SRv6 Policy Related APPDB Schemas
+The following piece is an example of VPNv4 route going via two SRv6 policies. One policy is for color 1 and the remote PE 2064:100::1d, and the other policy is for color 3 and the remote PE 2064:100::1e.
+
+```
+show ip route vrf PUBLIC-TC11 192.100.1.9/32 nexthop-group  
+Routing entry for 192.100.1.9/32
+  Known via "bgp", distance 20, metric 0, vrf PUBLIC-TC11, best
+  Last update 08:44:23 ago
+  Nexthop Group ID: 8661
+  PIC Context ID: 8657
+  * 2064:100::1d segment-list (vrf Default), label implicit-null, seg6local unspec unknown(seg6local_context2str), seg6 fd00:201:201:fff1:11::, weight 1, srv6tunnel(endpoint|color):2064:100::1d|1
+  * 2064:100::1e segment-list (vrf Default), label implicit-null, seg6local unspec unknown(seg6local_context2str), seg6 fd00:202:202:fff2:22::, weight 1, srv6tunnel(endpoint|color):2064:100::1e|3
+```
+
+## Route Schema
+VPN Route uses the existing route netlink format with pic_context_id added in TLV to pass route's information from zebra to fpmsyncd. Fpmsyncd would write this route in APPDB. In the example above, VPNv4 route points NHG with id 8661 for forwarding information and 8657 for PIC context, a.k.a VPN SID information.
+```
+127.0.0.1:6380> hgetall "ROUTE_TABLE:Vrf10000:192.100.1.9/32"
+ 1) "nexthop_group"
+ 2) "8661"
+ 3) "pic_context_id"
+ 4) "8657"
+ 5) "nexthop"
+ 6) "na"
+ 7) "ifname"
+ 8) "na"
+ 9) "vni_label"
+10) "na"
+11) "vpn_sid"
+12) "na"
+13) "segment"
+14) "na"
+15) "seg_src"
+16) "na"
+17) "blackhole"
+18) "false"
+```
+
+## PIC Contexts schema
+PIC context is mapped from a NHG in zebra by fpmsyncd during NHG handling. The main purpose for this object is to provide VPN sid.
+```
+ARISTA03T1# show nexthop-group rib  8657    
+ID: 8657 (zebra 0x562eba98bb00)
+     RefCnt: 0
+     segment_ref: 10
+     Uptime: 08:48:11
+     VRF: Default
+     Falgs: 0x403
+     Valid, Installed
+     SegDepends: (8583) (8658)
+           via 2064:100::1d segment-list  color 1 (vrf Default), label implicit-null, seg6local unspec unknown(seg6local_context2str), seg6 fd00:201:201:fff1:11::, weight 1
+           via 2064:100::1e segment-list  color 3 (vrf Default), label implicit-null, seg6local unspec unknown(seg6local_context2str), seg6 fd00:202:202:fff2:22::, weight 1
+     pic nhe:8661 
+```
+
+This PIC context would be stored in APPDB's PIC_CONTEXT_TABLE. The schema is the similar to current NEXTHOP_GROUP_TABLE's schema.
+```
+127.0.0.1:6380> hgetall "PIC_CONTEXT_TABLE:8657"
+1) "nexthop"
+2) "2064:100::1d,2064:100::1e"
+3) "vni_label"
+4) "na,na"
+5) "vpn_sid"
+6) "fd00:201:201:fff1:11::,fd00:202:202:fff2:22::"
+7) "seg_src"
+8) "2064:100::1f,2064:100::1f"
+```
```

Comment:
Does fpmsyncd create NHGroup table based on PIC_CONTEXT_TABLE?

Reply:
PIC_CONTEXT_TABLE is for VPN SID, not for TE/ Policy, which is discussed in https://github.com/eddieruan-alibaba/SONiC/blob/eruan-pic/doc/pic/bgp_pic_edge.md#high-level-design